### DISCUSSION TITLE
Edge hover styling

### DISCUSSION THREAD

**Discussion by gitHoussem** [bcakmakoglu/vue-flow#1063] [Q&A] [answered]
Hello, 
Quick question:
When I hover over an edge, I have a specific styling to make the stroke take an orange color. However, I am not able to make on edge move on top of all other overlaid edges so as it becomes visible. 
Here's a gif and some screenshots that better explain my issue: 
 
In the below gif, the first edge I hover over is the (apparently) a top one. So, when it becomes orange it hides the other bottom edges that are in black. However, the second hover on the other edge doesn't do that. 
![hover overlay](https://github.com/bcakmakoglu/vue-flow/assets/43743206/f8d56554-0921-424f-abc1-597c06a00c11)

In this screenshot I am hovering over the 'horizontal' edge, it becomes orange but the edge on top of it is still appearing in black. 
![image](https://github.com/bcakmakoglu/vue-flow/assets/43743206/cbbf204d-a87a-4fef-9f89-21d45bf5ed5d)

I apologize if the issue comes from my code and not the library, I am asking anyways because I tried playing with the z-index in css but couldn't make it work.

Thanks for your time.

**Comment by bcakmakoglu** [comment]
There a property called `edge.zIndex` that you can modify.

**Reply by gitHoussem** [reply]
Thanks for the quick answer. 
Where and how should this be used when hovering ? 
For context, I am using `:default-edge-options` prop in `<VueFlow>`  and a custom css on hover below: 
```css

  .vue-flow__edge-path:hover {
    stroke: orange !important;
    stroke-width: 2;
  }
```

**Reply by gitHoussem** [reply]
is there an "on-hover" event on edges so that I update the css dynamically or something like that ?

**Reply by bcakmakoglu** [reply, marked as answer]
```ts
onEdgeMouseEnter(({ edge }) => {
  edge.zIndex = 9999
  edge.style = {
    stroke: 'orange',
    strokeWidth: '2px',
  }
})

onEdgeMouseLeave(({ edge }) => {
  edge.zIndex = undefined
  edge.style = {}
})
```

**Reply by bcakmakoglu** [reply]
[Here's an example.](https://codesandbox.io/p/sandbox/headless-architecture-rz9mtn?file=%2Findex.html%3A1%2C1)

**Reply by gitHoussem** [reply]
Thank you so much man ! 
You rock !
